File: ios/Shared/YearFormatting.swift
import Foundation

/// Event-year display helpers.
///
/// SwiftUI `Text("\(2027)")` / `Text("… \(year)")` uses `LocalizedStringKey`, which
/// applies locale grouping → **"2,027"**. Always format years through here (or
/// `String(year)`) before putting them in UI copy.
enum YearFormatting {
    /// Plain digits only — never `"2,027"`.
    static func label(_ year: Int) -> String { String(year) }

    /// `"Rendezvous 2027"` style titles.
    static func rendezvousTitle(_ year: Int) -> String {
        "Rendezvous \(label(year))"
    }
}

File: android/core/network/src/main/kotlin/com/rendezvousil/core/network/YearFormatting.kt
package com.rendezvousil.core.network

/**
 * Event-year display helpers.
 *
 * Prefer these over raw Int interpolation in UI copy so years always show as
 * plain digits (mirrors iOS [YearFormatting] — SwiftUI locale grouping is the
 * main risk; keep Android consistent anyway).
 */
object YearFormatting {
    /** Plain digits only. */
    fun label(year: Int): String = year.toString()

    /** `"Rendezvous 2027"` style titles. */
    fun rendezvousTitle(year: Int): String = "Rendezvous ${label(year)}"
}
